src/test/java/com/commercetools/sync/products/utils/ProductUpdateActionUtilsTest.java

```diff
@@ -254,23 +284,106 @@ private void assertChangeMasterVariantEmptyErrorCatcher(final String productMock
     }
 
     @Test
-    public void buildAddVariantUpdateActionFromDraft_returnsProperties() throws Exception {
-        List<AttributeDraft> attributeList = Collections.emptyList();
-        List<PriceDraft> priceList = Collections.emptyList();
-        List<Image> imageList = Collections.emptyList();
-        ProductVariantDraftDsl draft = ProductVariantDraftBuilder.of()
-                .attributes(attributeList)
-                .prices(priceList)
-                .sku("testSKU")
-                .key("testKey")
-                .images(imageList)
-                .build();
-
-        AddVariant addVariant = buildAddVariantUpdateActionFromDraft(draft);
-        assertThat(addVariant.getAttributes()).isSameAs(attributeList);
-        assertThat(addVariant.getPrices()).isSameAs(priceList);
-        assertThat(addVariant.getSku()).isEqualTo("testSKU");
-        assertThat(addVariant.getKey()).isEqualTo("testKey");
-        assertThat(addVariant.getImages()).isSameAs(imageList);
+    public void buildAddVariantUpdateActionFromDraft_WithAttribsPricesAndImages_ShouldBuildCorrectAddVariantAction() {
+        // preparation
+        final List<AttributeDraft> attributeList = Collections.emptyList();
+        final List<PriceDraft> priceList = Collections.emptyList();
+        final List<Image> imageList = Collections.emptyList();
+        final ProductVariantDraft draft = ProductVariantDraftBuilder.of()
+                                                                    .attributes(attributeList)
+                                                                    .prices(priceList)
+                                                                    .sku("testSKU")
+                                                                    .key("testKey")
+                                                                    .images(imageList)
+                                                                    .build();
+
+        // test
+        final List<UpdateAction<Product>> result = buildAddVariantUpdateActionFromDraft(draft);
+
+        // assertion
+        assertThat(result).hasOnlyOneElementSatisfying(action -> {
+            assertThat(action).isInstanceOf(AddVariant.class);
+            final AddVariant addVariant = (AddVariant) action;
+            assertThat(addVariant.getAttributes()).isSameAs(attributeList);
+            assertThat(addVariant.getPrices()).isSameAs(priceList);
+            assertThat(addVariant.getSku()).isEqualTo("testSKU");
+            assertThat(addVariant.getKey()).isEqualTo("testKey");
+            assertThat(addVariant.getImages()).isSameAs(imageList);
+        });
+    }
+
+    @Test
+    public void buildAddVariantUpdateActionFromDraft_WithNoAssets_BuildsNoAddAssets() {
```

Comment:
Isn't this test already included in `buildAddVariantUpdateActionFromDraft_WithAttribsPricesAndImages_ShouldBuildCorrectAddVariantAction`?

Reply:
It's including but the assertions are different. The two tests are asserting different outcomes.

Reply:
But assets are also one of the property, right? It's different action now because it's a bug, but it should not be treated as different property.